SIL.Scripture/VerseRef.cs

```diff
@@ -1338,6 +1337,27 @@ static void ParseVerseNumber(string vNum, out int number, out string segment)
 			segment = vNum.Substring(j);
 		}
 
+		/// <summary>
+		/// Parses a verse string and gets the leading numeric portion as a number.
+		/// Functionally identical to Verse.Set for Roman numbers, made distinct to preserve USX standard.
+		/// </summary>
+		/// <returns><c>true</c> if the entire string could be parsed as a single,
+		/// simple verse number in any supported script; <c>false</c> if the verse string represented
+		/// a verse bridge, contained segment letters, or was invalid</returns>
+		public bool TrySetVerseUnicode(string value)
+		{
+			{
+				verse = !TryGetVerseNum(value, out short vNum, false) ? value.Replace(rtlMark, "") : null;
+				verseNum = vNum;
+				if (verseNum >= 0)
+					return true;
+
+				Trace.TraceWarning("Just failed to parse a verse number: " + value);
+				TryGetVerseNum(verse, out verseNum, false);
+				return false;
+			}
+		}
```

Comment:
This method is almost identical to the code in `Verse.set`. We should refactor this to a private method `private bool SetVerse(string value, bool romanOnly)` to remove the code duplication.

Reply:
Good point. Changed.